submariner/cable-driver-protocols-enhancement.md

```diff
@@ -0,0 +1,189 @@
+it will be difficult to achieve reliability and stability, especially if there is a complex topology of tunnels. In
+order to achieve such scales, using other L2/L3 VPN tech combined with an optional external IPSEC protection provides
+a better approach, because it allows the use of dynamic routing protocols instead of complex static routing. Furthermore,
+it allows the use of traffic engineering and Site to Site QoS which is impossible in the case of IPSEC Tunnel mode.
+* **Debuggability**: Tunnel mode IPSEC is a debugging nightmare (especially if it is offloaded in part or in full). There
+is no visibility of what goes into the VPN and once it has entered the tunnel it is encrypted and thus unsuitable for debugging.
+This problem goes away if the VPN is using standard L2/L3 VPN technology which provides suitable interfaces on which the system
+administrator can snoop.
+
+## Proposal
+
+### IP-in-IP Encapsulation
+
+This enhancement proposes to support IP-in-IP for un-encrypted connection between clusters. A
+new IP tunnelling interface `ipip-tunnel` will be added to the active gateway node of each cluster
+and it will act as the Tunnel End Point. The traffic which is destined to other clusters will be
+forwarded via this interface to the appropriate clusters.
+
+> **_NOTE:_** A single IP Tunnelling interface will be added, and tunnels managed using
+`# ip route add encap ip <args>` in order to avoid a point to point connection per tunnel.
+
+#### Design Details
+
+* A new interface "ipip-tunnel" will be added.
+* The IP tunnelling interface will be one-to-many, that is a single interface will be used to
+connect to all the joined clusters.
+* For the "ipip-tunnel" IP, the prefix "243" will be used, followed by the rest of the blocks
+from the endpoint private IP address. For example, if the private IP address is 10.2.96.1 the
+"ipip-tunnel" IP would be 243.2.96.1.
+* The routes will be added to forward all the Service and Pod CIDR traffic to the respective
+remote Tunnel IPs.
+* Table 150 shall be used to add the routes for inter-cluster communication.
+
+Example of routeing rules added by the IPTun driver on a gateway node to other nodes
+
+``` bash
+ip route add 10.2.0.0/16 encap ip id 100 dst 172.18.0.21 via 243.18.0.21 dev ipip0 table 100 metric 100 src 10.1.96.0
+ip route add 100.2.0.0/16 encap ip id 100 dst 172.18.0.21 via 243.18.0.21 dev ipip0 table 100 metric 100 src 10.1.96.0
+ip route add 10.3.0.0/16 encap ip id 100 dst 172.18.0.8 via 243.18.0.8 dev ipip0 table 100 metric 100 src 10.1.96.0
+ip route add 100.3.0.0/16 encap ip id 100 dst 172.18.0.8 via 243.18.0.8 dev ipip0 table 100 metric 100 src 10.1.96.0
+ip route add 10.4.0.0/16 encap ip id 100 dst 172.18.0.4 via 243.18.0.4 dev ipip0 table 100 metric 100 src 10.1.96.0
+ip route add 100.4.0.0/16 encap ip id 100 dst 172.18.0.4 via 243.18.0.4 dev ipip0 table 100 metric 100 src 10.1.96.0
+```
+
+and the resulting routes on the gateway node:
+
+```bash
```

Comment:
```suggestion
```text
```

This is actually text I think. As-is, the shell prompt is being syntax highlighted as bash (command is a comment). Same applies to a few other places below.

Reply:
Fixed to console in https://github.com/submariner-io/enhancements/pull/103